tests/k8s/tests/04-toservices-test.sh

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+
+# This tests:
+# - Kubernetes ToServices egress policies for headless services
+
+dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+source "${dir}/../helpers.bash"
+# dir might have been overwritten by helpers.bash
+dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+
+source "${dir}/../cluster/env.bash"
+
+TEST_NAME=$(get_filename_without_extension $0)
+LOGS_DIR="${dir}/cilium-files/${TEST_NAME}/logs"
+redirect_debug_logs ${LOGS_DIR}
+
+set -ex
+
+log "running test: $TEST_NAME"
+
+headless_dir="${dir}/deployments/headless"
+
+create_policy() {
+  if [[ "${k8s_version}" != 1.6.* ]]; then
+    kubectl create -f "${headless_dir}/policy.yaml"
+  else
+    kubectl create -f "${headless_dir}/policy-v1.yaml"
+  fi
+}
+
+delete_policy() {
+  if [[ "${k8s_version}" != 1.6.* ]]; then
+    kubectl delete -f "${headless_dir}/policy.yaml"
+  else
+    kubectl delete -f "${headless_dir}/policy-v1.yaml"
+  fi
+}
+
+cleanup (){
+  delete_policy
+  kubectl delete -f "${headless_dir}/service.yaml"
+  kubectl delete -f "${headless_dir}/pod.yaml"
+  kubectl delete -f "${headless_dir}/endpoint.yaml"
+}
+
+finish_test (){
+  log "finishing test: ${TEST_NAME}"
+  cleanup
+  gather_files ${TEST_NAME} k8s-tests
+  gather_k8s_logs "2" ${LOGS_DIR}
+  log "finished running test: $TEST_NAME"
+}
+trap finish_test exit
+
+NAMESPACE="kube-system"
+LOCAL_CILIUM_POD="$(kubectl get pods -n kube-system -o wide | grep $(hostname) | awk '{ print $1 }' | grep cilium)"
+
+expected_cidr='"198.49.23.144/32"'
+
+kubectl create -f "${headless_dir}/service.yaml"
+kubectl create -f "${headless_dir}/pod.yaml"
+
+kubectl get pods -o wide
+
+wait_for_running_pod toservices
+wait_for_cilium_ep_gen k8s ${NAMESPACE} ${LOCAL_CILIUM_POD}
+
+log "creating policy, then endpoint, checking if ip is in toCIDR rules of endpoint"
+
+create_policy
+kubectl create -f "${headless_dir}/endpoint.yaml"
+wait_for_cilium_ep_gen k8s ${NAMESPACE} ${LOCAL_CILIUM_POD}
+
+epID=$(kubectl exec -n kube-system ${LOCAL_CILIUM_POD} cilium endpoint list | awk '$2=="Enabled" { print $1 }')
+log "endpoint ID: ${epID}"
+
+cidr=$(kubectl exec -n kube-system ${LOCAL_CILIUM_POD} cilium endpoint get ${epID} | jq '.[0]["policy"]["cidr-policy"]["egress"][0]' -M)
```

Comment:
The test failed because `jq` is not installed on the VM, can't you workaround it with grep?

Reply:
I'm adding jq to the packer scripts

Reply:
Thanks!

Reply:
When will vms use new packer scripts?

Reply:
I'm a bit confused now as `jq` was already part of the packer build scripts.

Reply:
@ianvernon can you shed some light on it?